webapp/src/services/api.ts

```diff
@@ -224,16 +216,16 @@ export const api = createApi({
         "UtteranceCountPerFilter",
         "Utterances",
       ],
-      async onQueryStarted(
-        { persistentIds, newValue, ...args },
-        { dispatch, queryFulfilled }
-      ) {
+      async onQueryStarted({ body, ...args }, { dispatch, queryFulfilled }) {
         const patchResult = dispatch(
           api.util.updateQueryData("getUtterances", args, (draft) => {
             draft.utterances.forEach((utterance) => {
-              if (persistentIds.includes(utterance.persistentId)) {
-                utterance.dataAction = newValue;
-              }
+              return {
+                ...utterance,
+                ...body.filter(
+                  ({ persistentId }) => persistentId === utterance.persistentId
+                )[0],
```

Comment:
This `.filter(...)[0]` gives the same result as (but in a less efficient way than) `.find(...)`.
```suggestion
                ...body.find(
                  ({ persistentId }) => persistentId === utterance.persistentId
                ),
```